src/remote_types.rs

```diff
@@ -12,6 +12,17 @@ use std::collections::HashMap;
 use std::marker::PhantomData;
 use xxhash_rust::xxh32::xxh32;
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ClusterMetadataManifest {
+    pub version: u32, // This field is explicit in JSON, in serde it is the envelope version
```

Comment:
Does this comment make sense?  It's like the one on PartitionManifest, but I am guessing this new struct doesn't have json and serde encodings

Reply:
The struct does have the encodings; I made it an envelope type to 1. hopefully make it easier to reason about adding/removing fields, 2. open the door to eventually serializing the manifest, e.g. if in the recovery path we want to serialize the cluster metadata manifest.

In either case, I removed the comments here since they're note particularly useful.